gomock/argument_captor.go

```diff
@@ -0,0 +1,37 @@
+package gomock
+
+type ArgumentCaptor interface {
+	Matches(x interface{}) bool
+	Value() interface{}
+	AllValues() []interface{}
+}
+
+type argumentCaptor struct {
+	Matcher
+	values []interface{}
+}
+
+func (ac *argumentCaptor) Matches(x interface{}) bool {
+	ac.values = append(ac.values, x)
+	return ac.Matcher.Matches(x)
+}
+
+// If called multiple times, return the last argument value used
```

Comment:
Maybe just rename to LastValue() and drop the comment? Also the comment is a bit misleading. It says "if called multiple times" but it doesn't refer to calling of this Value() method.

Reply:
I think either way is a tradeoff here, `LastValue` vs `Value`. I would actually lean towards the original implementation of `Value` /w the documentation.  @cvgw WDYT?

Reply:
I would also be ok to go back to `Value()` and `AllValues()`.

Reply:
Please feel free to go back to the original implementation as @codyoss suggested.

Reply:
alright I'll do that